fedmsg_meta_fedora_infrastructure/fasshim.py

```diff
@@ -110,16 +111,22 @@ def make_fasjson_cache(**config):
         client = fasjson_client.Client(url=creds.get('base_url', default_url))
     except fasjson_client.errors.ClientSetupError as e:
         log.error("Something went wrong setting up fasjson client with error: %s" % e)
+        return {}
     try:
         response = client.list_users(page_size=1000)
         _add_to_cache(response.result)
         total_pages = response.page.get('total_pages')
-        for x in range(1, total_pages):
-            users = client.list_users(page_size=1000, page=x).result
+        # we want to start at the second page, since we already have the first
+        page_number = 2
+        while page_number < total_pages+1:
```

Comment:
For the record I would have used something like:
```python
page_number = 0
next_page_exists = True
while next_page_exists:
    page_number += 1
    response = client.list_users(page_size=1000, page=page_number)
    _add_to_cache(response.result)
    next_page_exists = page_number < response.page["total_pages"]
```
This way the only visible API call is in the while loop, you don't do one just for the 1st page and then more, so if you have to change the page_size one day you only do it once and you can't forget the other place.

I'm a very strong believer in the "don't repeat yourself" principle, and it influences my writing a lot, but your way of doing it works too. It was just a FYI :-)

Reply:
I added this because its better :)